test/selectBuildConfigs.test.ts

```diff
@@ -0,0 +1,157 @@
+        { workflowIndex: 0, buildConfig: 'output_location', expectedSelection: { line: 8, startChar: 27, endChar: 29 } },
+        { workflowIndex: 0, buildConfig: 'app_artifact_location', expectedSelection: undefined },
+
+        { workflowIndex: 1, buildConfig: 'api_location', expectedSelection: { line: 7, startChar: 24, endChar: 38 } },
+        { workflowIndex: 1, buildConfig: 'app_location', expectedSelection: { line: 6, startChar: 24, endChar: 38 } },
+        { workflowIndex: 1, buildConfig: 'output_location', expectedSelection: undefined },
+        { workflowIndex: 1, buildConfig: 'app_artifact_location', expectedSelection: { line: 8, startChar: 33, endChar: 54 }},
+
+        { workflowIndex: 2, buildConfig: 'api_location', expectedSelection: undefined },
+        { workflowIndex: 2, buildConfig: 'app_location', expectedSelection: undefined },
+        { workflowIndex: 2, buildConfig: 'output_location', expectedSelection: undefined },
+        { workflowIndex: 2, buildConfig: 'app_artifact_location', expectedSelection: undefined },
+
+        { workflowIndex: 3, buildConfig: 'api_location', expectedSelection: { line: 30, startChar: 24, endChar: 39 }},
+        { workflowIndex: 3, buildConfig: 'app_location', expectedSelection: { line: 29, startChar: 24, endChar: 57 }},
+        { workflowIndex: 3, buildConfig: 'output_location', expectedSelection: { line: 31, startChar: 27, endChar: 54 }},
+        { workflowIndex: 3, buildConfig: 'app_artifact_location', expectedSelection: undefined },
+    ];
+
+    const workflowProvider: TextDocumentContentProvider = new (class implements TextDocumentContentProvider {
+        provideTextDocumentContent(uri: Uri): string {
+            return workflows[parseInt(uri.path)];
+        }
+    })();
+    const scheme: string = 'testWorkflows';
+    workspace.registerTextDocumentContentProvider(scheme, workflowProvider);
+
+    for (const testCase of testCases) {
+        const title: string = `Workflow ${testCase.workflowIndex}: ${testCase.buildConfig}`;
+
+        test(title, async () => {
+            const uri: Uri = Uri.parse(`${scheme}:${testCase.workflowIndex}`);
+            const configDocument: TextDocument = await workspace.openTextDocument(uri);
+            const selection: Range | undefined = await tryGetSelection(configDocument, testCase.buildConfig);
+            let expectedSelection: Range | undefined;
+
+            if (testCase.expectedSelection) {
+                const expectedStart: Position = new Position(testCase.expectedSelection.line, testCase.expectedSelection.startChar);
+                const expectedEnd: Position = new Position(testCase.expectedSelection.line, testCase.expectedSelection.endChar);
+                expectedSelection = new Range(expectedStart, expectedEnd);
+            }
+
+            assert.ok(expectedSelection && selection?.isEqual(expectedSelection) || selection === expectedSelection, 'Actual and expected selections do not match');
```

Comment:
You should be able to use `assert.strictEquals()` here.  Using `assert.ok` seems kind of roundabout here.

Reply:
I went with `assert.ok` because comparing two ranges uses their `isEqual()` function. I updated the assert to hopefully make it look more purposeful

Reply:
Ahh, okay, makse sense.